### PR TITLE
add lely-core/2.3.2

### PR DESCRIPTION
**Pull request by uyha**
Specify library name and version:  **lely-core/2.3.1**

`lely-core` is a library for interacting with CANopen and it is not available on Conan Center yet, so I want to have it available there.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/lely-core/all/conanfile.py
Comment (RubenRBS) [inline recipes/lely-core/all/conanfile.py:26]: I know it's like this upstream too, but I've found a typo (Unless hih actually means something in the domain 🤷‍♂️!):
```suggestion
    description = "The Lely core libraries are a collection of C and C++ libraries and tools, providing high-performance I/O and sensor/actuator control for robotics and IoT applications. The libraries are cross-platform and have few dependencies. They can be even be used on bare-metal microcontrollers with as little as 32 kB RAM."
```

File: recipes/lely-core/all/conanfile.py
Comment (RubenRBS) [inline recipes/lely-core/all/conanfile.py:221]: I love how clean this looks!

File: recipes/lely-core/all/conanfile.py
Comment (RubenRBS) [inline recipes/lely-core/all/conanfile.py:122]: Can you add some context to the error here? (Like the comment above) UX would be nicer in case someone triggers this locally

**Review by RubenRBS — DISMISSED** [review]
Thanks for taking the time to add lely-code, the code looks great as far as I can tell :) My review is only super minor nitpicks

File: recipes/lely-core/all/conanfile.py
Comment (uyha) [reply recipes/lely-core/all/conanfile.py:26]: yeah, i think that's just a typo

File: recipes/lely-core/all/conanfile.py
Comment (uyha) [reply recipes/lely-core/all/conanfile.py:221]: thank you

File: recipes/lely-core/all/conanfile.py
Comment (uilianries) [inline recipes/lely-core/all/conanfile.py:24]:
```suggestion
    license = "Apache-2.0"
```
Please, always use license SPDX short-name only. https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#license-attribute

File: recipes/lely-core/all/conanfile.py
Comment (uilianries) [inline recipes/lely-core/all/conanfile.py:121]:
```suggestion
                f"{self.ref} is only compatible with Linux. "
```
Conan version sounds like Conan client version. Please, use the package reference instead.

File: recipes/lely-core/all/conanfile.py
Comment (uilianries) [inline recipes/lely-core/all/conanfile.py:127]:
```suggestion
                f"{self.ref} can only be compiled with GCC currently"
```

File: recipes/lely-core/all/conanfile.py
Comment (uilianries) [inline recipes/lely-core/all/conanfile.py:151]: Please, move it to AutotoolsToolchain with configure_args: https://docs.conan.io/en/latest/reference/conanfile/tools/gnu/autotoolstoolchain.html#attributes

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution!

**Comment by uyha** [comment]
@uilianries @RubenRBS @prince-chrismc I'm really sorry for tagging and pushing, but I need to use this library in a project and would like to import it through Conan, and I think this package is ready to be merged in. Could you review it again?

File: recipes/lely-core/all/conandata.yml
Comment (uilianries) [inline recipes/lely-core/all/conandata.yml:2]: You could make available only 2.3.2 if you want too. In case someone need 2.3.1, we can add in another PR. Less versions, less maintenance 😄

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by prince-chrismc** [comment]
> I'm really sorry for tagging and pushing, but I need to use this library in a project and would like to import it through Conan, and I think this package is ready to be merged in. Could you review it again?

You can check my bot https://github.com/prince-chrismc/conan-center-index-pending-review/issues/1 so learn about the status of all PRs there are fun graphs at the bottom

it's currently taking ~16 days for PRs to merge so you'll need to be patient as we have a huge backlog from the holidays

File: recipes/lely-core/all/conandata.yml
Comment (prince-chrismc) [reply recipes/lely-core/all/conandata.yml:2]: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/sources_and_patches.md#adding-old-versions

File: recipes/lely-core/all/test_package/CMakeLists.txt
Comment (prince-chrismc) [inline recipes/lely-core/all/test_package/CMakeLists.txt:8]: pthread should be added as a system library in the recipe

File: recipes/lely-core/all/conandata.yml
Comment (uyha) [reply recipes/lely-core/all/conandata.yml:2]: yeah, i also thought that made sense and was about to remove it, but didn't want to wait for a long for another review. I will remove it.

**Comment by uyha** [comment]
> > I'm really sorry for tagging and pushing, but I need to use this library in a project and would like to import it through Conan, and I think this package is ready to be merged in. Could you review it again?
> 
> You can check my bot [prince-chrismc/conan-center-index-pending-review#1](https://github.com/prince-chrismc/conan-center-index-pending-review/issues/1) so learn about the status of all PRs there are fun graphs at the bottom
> 
> it's currently taking ~16 days for PRs to merge so you'll need to be patient as we have a huge backlog from the holidays

I checked that, but I became anxious (due to no review, and pressure to use it) so I am really sorry again for this.

**Comment by RubenRBS** [comment]
No worries, we understand that it's hard when you feel like your PR gets ignored, but as Chris said, it's not the case! We're just running a bit behind due to the holidays :)

File: recipes/lely-core/all/test_package/CMakeLists.txt
Comment (uyha) [reply recipes/lely-core/all/test_package/CMakeLists.txt]: pthread is specified as a system library at line 229 in conanfile.py, i removed the `Threads::Threads` linking

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**
